src/sds.c

```diff
@@ -1136,21 +1159,31 @@ static int sdsparsearg(const char *arg, unsigned int *len, char *dst) {
  * extra space.
  */
 sds *sdssplitargs(const char *line, int *argc) {
+    return sdsnsplitargs(line, -1, argc);
+}
+
+/* Split a line into arguments, like sdssplitargs(), but with an explicit length parameter.
+ * The number of arguments is stored into *argc, and an array of sds is returned.
+ * The caller should free the resulting array with sdsfreesplitres().
+ */
+sds *sdsnsplitargs(const char *line, int len, int *argc) {
```

Comment:
Maybe it would be better to avoid integer type for len?
I mean we could do a:

```
sds *sdsnsplitargs(const char *line, const char *end, int *argc);
```

and implement:

```
sds *sdssplitargs(const char *line, int *argc) {
    return sdsnsplitargs_internal(line, NULL, argc);
}

sds *sdsnsplitargs(const char *line, size_t len, int *argc) {
    const char *end = len < 0 ? NULL : line + len;
    return sdsnsplitargs_internal(line, end, argc);
}
```